agent/eventhandler/task_handler.go

```diff
@@ -347,11 +349,21 @@ func (taskEvents *taskSendableEvents) submitFirstEvent(handler *TaskHandler, bac
 	} else if event.taskShouldBeSent() {
 		if err := event.send(sendTaskStatusToECS, setTaskChangeSent, "task",
 			handler.client, eventToSubmit, handler.stateSaver, backoff, taskEvents); err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == ecs.ErrCodeInvalidParameterException {
```

Comment:
can you move this logic into its own method? you can then avoid code duplication here and in line 362:
```golang
if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == ecs.ErrCodeInvalidParameterException
```

Reply:
Ok. What about further generalizing it into IsAwsErrAndEqualToErrCode(error, errorCode) and put this function at agent/utils/error.go? same logic is used in agent/app/agent.go line 470 and we will use more similar logics in the next pr for RegisterContainerInstance API:
```golang
if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == ecs.ErrCodeInvalidParameterException {
```

Reply:
You can move the error check into a method. Also can you add another method that handles this error type in this file?

Reply:
OK, sure.